source/PlayerInfo.cpp

```diff
@@ -1344,8 +1344,18 @@ void PlayerInfo::ReorderShip(int fromIndex, int toIndex)
 
 	// Reorder the list.
 	shared_ptr<Ship> ship = ships[fromIndex];
+	auto oldFirstShip = ships[0];
 	ships.erase(ships.begin() + fromIndex);
 	ships.insert(ships.begin() + toIndex, ship);
+	auto newFirstShip = ships[0];
+	// Check if the ship in the first position can be a flagship and is in the current system.
+	auto firstShip = ships.begin();
+	if(newFirstShip != oldFirstShip && Preferences::Has("Automatically unpark flagship")
+			&& (*firstShip)->CanBeFlagship() && (*firstShip)->GetSystem() == system && newFirstShip->IsParked())
+	{
+		newFirstShip->SetIsParked(false);
+		oldFirstShip->SetIsParked(true);
+	}
```

Comment:
Since this code is duplicated, could you perhaps move this to a function in the namespace? Also, I don't think there is a need for a separate `firstShip` and `newFirstShip`.

Putting something like this in the namespace should be sufficient:
```
	void HandleFlagshipParking(Ship *oldFirstShip, Ship *newFirstShip, const System *system)
	{
		if(newFirstShip != oldFirstShip && Preferences::Has("Automatically unpark flagship")
				&& newFirstShip->CanBeFlagship() && newFirstShip->GetSystem() == system && newFirstShip->IsParked())
		{
			newFirstShip->SetIsParked(false);
			oldFirstShip->SetIsParked(true);
		}
	}
```
and then you can call that function here.

Reply:
It does work. Thanks for the advice :) But the function call in "ReorderShip" doenst look nice to me. &(*oldFirstShip) seems to be needed for the compiler. It was the only way I got it to work. Maybe you know a better looking way for passing the ships?